src/main/java/edu/hm/hafner/coverage/Node.java

```diff
@@ -647,11 +647,14 @@ private void mergeChildren(final Node other) {
     private void mergeValues(final Value otherValue) {
         if (getMetricsOfValues().anyMatch(v -> v.equals(otherValue.getMetric()))) {
             var old = getValueOf(otherValue.getMetric());
-            if (hasChildren()) {
-                replaceValue(old.add(otherValue));
+            try {
+                var max = old.max(otherValue);
```

Comment:
With this functionality it's still a replace right? It just takes the coverage that's higer.
If there is a file with the functions `a` and `b`, coverage X covers func `a` and coverage Y covers function `b`, it's 50% overall coverage instead of 100% right?

Reply:
It is still not clear on how to merge inner nodes 😞 

The code works for leaves but not for inner nodes yet. I can drop the details of inner nodes and compute them on my own but that will cause other problems... 

Do you have an idea on how to implement this correctly? I do not see a solution for this problem.

Reply:
I don't have much experience with coverage reports, but by just looking at the cobertura one, I would expect that the lines should be merged (identified by number and branch) and `hits > 0` should be preferred

Reply:
After adding some more tests it looks that the merge using `max` is working correctly. 

> With this functionality it's still a replace right? It just takes the coverage that's higer.
If there is a file with the functions a and b, coverage X covers func a and coverage Y covers function b, it's 50% overall coverage instead of 100% right?

No, since each file has a coverage for X and Y. So X contains a=100% and b=0%, and Y contains a=0% and b=100%, then the max will produce a=100% and b=100%.

Reply:
I mean it's more correct than before.
Lets say we have function A.
```ts
function A(flag: boolean) {
  if(flag) {
    console.log('true');
  } else {
   console.log('false');
  }
}
```
If you have two separate coverage reports, one with the file `x.spec.test` that tests `A` only with `true` and one with the file `y.spec.test` that tests `A` only with `false`, then both of them would have the same coverage (about 50%) and the code-coverage-api would return 50% right?

Reply:
> I mean it's more correct than before. Lets say we have function A.
> 
> ```ts
> function A(flag: boolean) {
>   if(flag) {
>     console.log('true');
>   } else {
>    console.log('false');
>   }
> }
> ```
> 
> If you have two separate coverage reports, one with the file `x.spec.test` that tests `A` only with `true` and one with the file `y.spec.test` that tests `A` only with `false`, then both of them would have the same coverage (about 50%) and the code-coverage-api would return 50% right?

No, the coverage is not measured per method, it is line based. The coverage per method is computed from the line coverage. So you should get a 100% coverage in total.